aioredis/pool.py

```diff
@@ -160,6 +161,8 @@ def closed(self):
     @asyncio.coroutine
     def wait_closed(self):
         """Wait until pool gets closed."""
+        if self._close_waiter is None:
+            self._close_waiter = async_task(self._do_close(), loop=self._loop)
```

Comment:
Otherwise you should add check `if self._close_waiter is None` in `close()` method.
This also applies to sentinel pool.